app/web/service/search.ts

```diff
@@ -98,12 +98,20 @@ export async function EventSearch({
     req.setPast(pastEvents);
   }
   if (typeof searchLocation !== "string") {
-    const location = new RectArea();
-    location.setLatMin(searchLocation?.bbox[1] || 0);
-    location.setLatMax(searchLocation?.bbox[3] || 0);
-    location.setLngMin(searchLocation?.bbox[0] || 0);
-    location.setLngMax(searchLocation?.bbox[2] || 0);
-    req.setSearchInRectangle(location);
+    // If it's a region (i.e. "France" or "United States") use query search by name
+    // This will search for events in the region by that name
+    if (searchLocation?.isRegion) {
+      req.setQuery(new StringValue().setValue(searchLocation.name));
```

Comment:
this works by I'm trying to understand why... this simply does something like a regex? like if the searched name is inside the event location name?

Reply:
explain before merging pls, might be useful for the map in the future (even if not want to understand it :D)

Reply:
Sure! It goes like this:

- We type a region like "France" in the search box
- That gets set in `LocationAutocomplete` component
- `LocationAutocomplete` component sends it in an api call to openstreetmap via the `useGeocodeQuery` hook
- Openstreetmap processes it, and gives us back the `GeocodeResult` object, standardizing the name, simplified name, bbox, lat and longitude, etc.
- We send that to our backend to to the `EventSearch` function to try to find matching events
- The backend checks what type of search: `search_in_area`, `search_in_rectangle` or `query` in this case. We define which search in the `search.ts` file with a grpc request
- We decide what to pass in the query. I am passing the location `name` passed back from the openstreet map response. We could also send `simplifiedName`, or something else.
- The backend checks if we want to search only the event title, or the event title, content AND address
- I chose the one with title, content and address
- Our event addresses have the city, state, country in them, that's why it works to search by country or city, as that's always in there. When you create an event, it also standardizes it with `LocationAutocomplete` via the openstreetmap api, so it's pretty safe that it will always match. So no chance someone makes a typo, leaves out the city or country, etc.

<img width="381" alt="Screenshot 2024-10-24 at 7 30 02 PM" src="https://github.com/user-attachments/assets/0a02365a-839f-4843-b2d4-1c868d6b283a">

The ONLY thing I could think that might go wrong...since the query also searches the content of the event, if the user adds the name of a different city in the event description, that might make it show up in a search.

So say I make an event in Paris, France, but in the description I write, "David and I met in Madrid and decided to organize this event together", the event might also show up when searching Madrid.

I think that's a rare edge case though, so it's probably okay.

Looks like @spreeni wrote this code on the backend, please correct me if I got anything wrong!